File: BensBible2/Models/HighlightColor.swift
import SwiftUI

enum HighlightColor: String, CaseIterable, Identifiable {
    case yellow
    case green
    case blue
    case pink
    case orange

    var id: String { rawValue }

    var color: Color {
        switch self {
        case .yellow: Color.yellow.opacity(0.33)
        case .green: Color.green.opacity(0.30)
        case .blue: Color.blue.opacity(0.30)
        case .pink: Color.pink.opacity(0.30)
        case .orange: Color.orange.opacity(0.33)
        }
    }

    var solidColor: Color {
        switch self {
        case .yellow: .yellow
        case .green: .green
        case .blue: .blue
        case .pink: .pink
        case .orange: .orange
        }
    }

    var displayName: String { rawValue.capitalized }
}

File: android/app/src/main/java/com/bensbible/app/model/HighlightColor.kt
package com.bensbible.app.model

import androidx.compose.ui.graphics.Color

enum class HighlightColor(val rawValue: String) {
    YELLOW("yellow"),
    GREEN("green"),
    BLUE("blue"),
    PINK("pink"),
    ORANGE("orange");

    val color: Color
        get() = when (this) {
            YELLOW -> Color(0x54FFEB3B)
            GREEN -> Color(0x4D4CAF50)
            BLUE -> Color(0x4D2196F3)
            PINK -> Color(0x4DE91E63)
            ORANGE -> Color(0x54FF9800)
        }

    val solidColor: Color
        get() = when (this) {
            YELLOW -> Color(0xFFFFEB3B)
            GREEN -> Color(0xFF4CAF50)
            BLUE -> Color(0xFF2196F3)
            PINK -> Color(0xFFE91E63)
            ORANGE -> Color(0xFFFF9800)
        }

    val displayName: String get() = rawValue.replaceFirstChar { it.uppercase() }

    companion object {
        fun fromRawValue(raw: String?): HighlightColor? {
            if (raw == null) return null
            return entries.find { it.rawValue == raw }
        }
    }
}
